### DISCUSSION TITLE
Fast-DDS/src/cpp/rtps/transport/TCPChannelResource.h:279:45 error！

### DISCUSSION THREAD

**Discussion by kait1234** [eProsima/Fast-DDS#5371] [Support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html#fast-dds-gen-installation，
using title is 3.1.4.1. Local installation，complie Fast-DDS。

### Current behavior

when complie Fast-DDS,some error show:
<img width="649" alt="bug" src="https://github.com/user-attachments/assets/7f062512-b374-4153-a70e-b5a0c82703b1">
how to do？

### Steps to reproduce

3.1.4.1. Local installation，in three step：3.Once all dependencies are installed, install eProsima Fast DDS
`cmake --build . --target install`
then compile stop。

### Fast DDS version/commit

latest，V3.1.0 2024-10-17

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

environment  is：  linux that base on centos7,x86 64 cpu。

**Comment by kait1234** [comment]
update Asio to 1.30.2,then compile completed!

**Comment by JesusPoderoso** [comment]
Hi @kait1234 , thanks for your report.
The required Asio version to be installed is v1.31.0 (check [dependencies section](https://fast-dds.docs.eprosima.com/en/stable/notes/versions.html#library-dependencies)).
You can also build Fast DDS with third-party dependencies, so it is not required to install Asio. This can be done using [these CMake options](https://fast-dds.docs.eprosima.com/en/stable/installation/configuration/cmake_options.html#third-party-libraries-options).

I am moving this thread to the corresponding support discussions forum as per the [Fast DDS contributing guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#issues-and-support).

_Edit: also bear in mind that CentOS 7 is not part of [Fast DDS supported platforms](https://fast-dds.docs.eprosima.com/en/stable/notes/versions.html#build-system-dependencies), so we cannot ensure that CentOS 7 Asio distribution is the minimum required._

**Comment by shynur** [comment]
> also bear in mind that CentOS 7 is not part of [Fast DDS supported platforms](https://fast-dds.docs.eprosima.com/en/stable/notes/versions.html#build-system-dependencies), so we cannot ensure that CentOS 7 Asio distribution is the minimum required.

Nor Ubuntu 18.04 ?
